Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -45,31 +48,52 @@ import SwiftUI
 ///     }
 /// }
 /// ```
+///
+/// ## Topics
+/// - ``init()``
+/// ### Navigating
+/// - ``nextStep()``
+/// ### Manipulating the Stack
+/// - ``removeLast()``
 @MainActor
 @Observable
 public class OnboardingNavigationPath {
-    /// Internal SwiftUI `NavigationPath` that serves as the source of truth for the navigation state.
-    /// Holds elements of type `OnboardingStepIdentifier` which identify the individual onboarding steps.
-    var path = NavigationPath()
+    enum StepReference {
+        case viewType(any View.Type)
+        case identifier(any Hashable)
+    }
+    
+    /// The actual path of onboarding steps currently presented.
+    var path: [OnboardingStepIdentifier] = [] {
+        didSet {
+            // Remove dismissed custom steps when navigating backwards
+            let removedSteps = oldValue.filter { !path.contains($0) }
+            for step in removedSteps where step.isCustom {
+                customOnboardingSteps.removeValue(forKey: step)
+            }
+        }
+    }
     /// Boolean binding that is injected via the ``OnboardingStack``.
     /// Indicates if the onboarding flow is completed, meaning the last view declared within the ``OnboardingStack`` is completed.
-    private let complete: Binding<Bool>?
+    private var isComplete: Binding<Bool>?
 
     /// Stores all onboarding views as declared within the ``OnboardingStack`` and keep them in order.
     private var onboardingSteps: OrderedDictionary<OnboardingStepIdentifier, any View> = [:]
     /// Stores all custom onboarding views that are appended to the `OnboardingNavigationPath`
-    /// via the ``append(customView:)`` or ``append(customViewInit:)`` instance methods
+    /// via the ``append(customView:)``  instance methods
     private var customOnboardingSteps: [OnboardingStepIdentifier: any View] = [:]
+    /// Indicates whether the Path's ``OnboardingNavigationPath/configure`` function has been called at least once.
+    private(set) var didConfigure = false
```

Comment:
Is there any chance that we can work around this entire need to configure the `OnboardingNavigationPath` and keep a state of that?

Reply:
the use case here is situations where the `OnboardingStack` gets fully reloaded (because of a full reload of the entire view hierarchy it belongs to) and is using an externally-owned path, in which case we don't want to fully reconfigure the path.